interfaces/builtin/socketcan_control_test.go

```diff
@@ -0,0 +1,111 @@
+	"github.com/snapcore/snapd/interfaces/builtin"
+	"github.com/snapcore/snapd/interfaces/seccomp"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type SocketCanControlInterfaceSuite struct {
+	iface    interfaces.Interface
+	slotInfo *snap.SlotInfo
+	slot     *interfaces.ConnectedSlot
+	plugInfo *snap.PlugInfo
+	plug     *interfaces.ConnectedPlug
+}
+
+var _ = Suite(&SocketCanControlInterfaceSuite{
+	iface: builtin.MustInterface("socketcan-control"),
+})
+
+const socketCanControlConsumerYaml = `name: consumer
+version: 0
+apps:
+ app:
+  plugs: [socketcan-control]
+`
+
+const socketCanControlCoreYaml = `name: core
+version: 0
+type: os
+slots:
+  socketcan-control:
+`
+
+func (s *SocketCanControlInterfaceSuite) SetUpTest(c *C) {
+	s.plug, s.plugInfo = MockConnectedPlug(c, socketCanControlConsumerYaml, nil, "socketcan-control")
+	s.slot, s.slotInfo = MockConnectedSlot(c, socketCanControlCoreYaml, nil, "socketcan-control")
+}
+
+func (s *SocketCanControlInterfaceSuite) TestName(c *C) {
+	c.Assert(s.iface.Name(), Equals, "socketcan-control")
+}
+
+func (s *SocketCanControlInterfaceSuite) TestSanitizeSlot(c *C) {
+	c.Assert(interfaces.BeforePrepareSlot(s.iface, s.slotInfo), IsNil)
+	slot := &snap.SlotInfo{
+		Snap:      &snap.Info{SuggestedName: "some-snap"},
+		Name:      "socketcan-control",
+		Interface: "socketcan-control",
+	}
+	c.Assert(interfaces.BeforePrepareSlot(s.iface, slot), ErrorMatches,
+		"socketcan-control slots are reserved for the core snap")
+}
+
+func (s *SocketCanControlInterfaceSuite) TestSanitizePlug(c *C) {
+	c.Assert(interfaces.BeforePreparePlug(s.iface, s.plugInfo), IsNil)
+}
+
+func (s *SocketCanControlInterfaceSuite) TestAppArmorSpec(c *C) {
+	spec := &apparmor.Specification{}
+	c.Assert(spec.AddConnectedPlug(s.iface, s.plug, s.slot), IsNil)
+	c.Assert(spec.SecurityTags(), DeepEquals, []string{"snap.consumer.app"})
+	c.Assert(spec.SnippetForTag("snap.consumer.app"), testutil.Contains, "network can,\n")
+	c.Assert(spec.SnippetForTag("snap.consumer.app"), testutil.Contains, "capability net_admin,\n")
+}
+
+func (s *SocketCanControlInterfaceSuite) TestSecCompSpec(c *C) {
+	spec := &seccomp.Specification{}
+	c.Assert(spec.AddConnectedPlug(s.iface, s.plug, s.slot), IsNil)
+	c.Assert(spec.SecurityTags(), DeepEquals, []string{"snap.consumer.app"})
+	c.Assert(spec.SnippetForTag("snap.consumer.app"), testutil.Contains, "socket AF_NETLINK - NETLINK_ROUTE\n")
+}
+
+func (s *SocketCanControlInterfaceSuite) TestStaticInfo(c *C) {
+	si := interfaces.StaticInfoOf(s.iface)
+	c.Assert(si.ImplicitOnCore, Equals, true)
+	c.Assert(si.ImplicitOnClassic, Equals, true)
+	c.Assert(si.Summary, Equals, `allows configuration and use of SocketCAN network interfaces`)
+	c.Assert(si.BaseDeclarationSlots, testutil.Contains, "socketcan-control")
+}
+
+func (s *SocketCanControlInterfaceSuite) TestAutoConnect(c *C) {
+	c.Assert(s.iface.AutoConnect(s.plugInfo, s.slotInfo), Equals, true)
+}
```

Comment:
These days TestAutoConnect can be omitted (this is tested via interfaces/policy)